kuka_eki_hw_interface/include/kuka_eki_hw_interface/kuka_eki_hw_interface.h

```diff
@@ -38,10 +38,10 @@
 #include <vector>
 #include <string>
 
-#include <boost/asio.hpp>
-
 #include <ros/ros.h>
 #include <controller_manager/controller_manager.h>
+
+#include <boost/asio.hpp>
```

Comment:
Is this change needed to work-around `boost/asio.hpp` including `Windows.h` somewhere and then conflicting with the `ERROR` `enum class` in `ros_control`'s `robot_hw.h` (or something similar)?

It appears we're running into *something* which causes problems with the `ERROR` `#define` in https://github.com/ros-industrial/abb_robot_driver/issues/17.

Reply:
Yes, I can add a comment for that.
It's really unfortunate that #defines are not namespaced, and that 30 years ago someone thought #define ERROR was a good idea.

Reply:
Yes, that's what I was thinking as well :)

The strange thing is: some users of `abb_robot_driver` report this problem with ROS Melodic and ROS Noetic, while the main developer (@jontje) doesn't run into it (on Melodic).